### ISSUE TITLE
Load Saved Weights

### ISSUE THREAD

**Issue by quant-exchange** [ourownstory/neural_prophet#1542]
I notice this merge from the past: https://github.com/ourownstory/neural_prophet/pull/691/files

Just to make sure I understand here: I'm not seeing these methods in the forecaster.py any longer. Is there an official way to save the weights to file and load those on next run so NP isn't being complete new instantiation each time, therefore outside of the epoch window; there's no continuous machine learning capable with the current library?

If this is the case, I wouldn't mind figuring out how to make a PR to get a solve if possible.

**Comment by quant-exchange** [comment]
After doing some low-level weight save testing; I did notice some results close to NP results on accuracy; however, the weights also make 'consistency' in forecast more difficult form my testing so far. In general; I'm sure weight save and load is a bigger topic in forecasting (especially volatile datasets) than just implementing it like in standard PyTorch workflows.

**Comment by ourownstory** [comment]
> I'm not seeing these methods in the forecaster.py any longer.

The save and load methods have been slightly improved and now live here: https://github.com/ourownstory/neural_prophet/blob/dbd045863a23440763800fc463900011cc984035/neuralprophet/utils.py#L25

**Comment by ourownstory** [comment]
> After doing some low-level weight save testing; I did notice some results close to NP results on accuracy; however, the weights also make 'consistency' in forecast more difficult from my testing so far.

I would love to learn more about what you did and found here!

**Comment by ourownstory** [comment]
> continuous machine learning

We currently support saving/loading of the entire object (via torch).

If you are looking to train a model, then re-train/fine-tune the already trained model, that is currently not supported, but should be possible to support with a limited set of changes. It mainly requires adjusting some dataset and model initialization checks, and adjusting the re-training schedule. I'd be happy to support you if you decide to add this with a PR!

If you were planning to only save the weights and reload the weights, without all python/torch objects, that might be a larger project requiring sophisticated save and load methods that are highly dependent on many areas of the code.

**Comment by ourownstory** [comment]
Please feel free creating a PR that does away with this assertion: https://github.com/ourownstory/neural_prophet/blob/dbd045863a23440763800fc463900011cc984035/neuralprophet/forecaster.py#L963

Evidently, the PR will also need to be able to support the new logic of having a pre-initialized model now being trained again, including adequate test cases. Happy to help if you get stuck!

**Comment by quant-exchange** [comment]
Thanks! I'm going to be deep diving into this further and will review some test results. Will plan on making a PR in the near future for this!

**Comment by abdulmominseo** [comment]
My 52-Week Weight Loss Journey Unveiled | Ian Melton's weight loss story
Rating-⭐⭐⭐⭐⭐
[Full video](https://youtu.be/KTS0AXO3YHY?si=mdUW6BxiTfGWeibd)

**Comment by ourownstory** [comment]
Excellent @quant-exchange! Looking forward to your PR.

**Comment by ourownstory** [comment]
Addressing this will also require https://github.com/ourownstory/neural_prophet/issues/828
